### PR TITLE
[WEB-2273] Chore: header UI

### PR DESCRIPTION
**Pull request by gakshita**
**Summary**
- Implemented a global header component supporting multiple variants
- Wrapped all headers inside this component

[[WEB-2273]](https://app.plane.so/plane/projects/02c3e1d5-d7e2-401d-a773-45ecba45d745/issues/e0c02060-1e1d-4efc-abf2-34b4d12a4ada)

## Summary by CodeRabbit

- **New Features**
	- Introduced a customizable `CustomHeader` component for improved header layout across various components.
	- Added enhanced responsive design options with new utility classes in Tailwind CSS.

- **Bug Fixes**
	- Adjusted class names for better vertical alignment and layout consistency across multiple components.

- **Documentation**
	- Updated documentation to reflect new component structures and usage of `CustomHeader`.

- **Chores**
	- Refactored existing components to utilize the new `CustomHeader`, improving modularity and maintainability.
	- Removed unused props and commented out features to streamline component interfaces.

### CODE REVIEW

File: packages/ui/src/row/row.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { cn } from "../../helpers";
+import { ERowVariant, rowStyle, TRowVariant } from "./helper";
+
+export interface CustomRowProps extends React.HTMLAttributes<HTMLDivElement> {
+  variant?: TRowVariant;
+  className?: string;
+  children: React.ReactNode;
+}
+
+const CustomRow = React.forwardRef<HTMLDivElement, CustomRowProps>((props, ref) => {
```
Comment (SatishGandham) [inline packages/ui/src/row/row.tsx:11]: Call it row, and also can we keep the helpers and types in this file and the component is pretty simple and the information in helpers is key to understand the component.

File: packages/ui/src/header/helper.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+export type THeaderVariant = "primary" | "secondary" | "ternary";
```
Comment (rahulramesha) [inline packages/ui/src/header/helper.tsx:1]: Minor, you can use the enum's values as options here

File: packages/ui/src/header/helper.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+export enum EHeaderVariant {
+  primary = "primary",
+  secondary = "secondary",
+  ternary = "ternary",
+}
+export interface IHeaderProperties {
+  [key: string]: string;
+}
+export const headerStyle: IHeaderProperties = {
+  primary:
```
Comment (rahulramesha) [inline packages/ui/src/header/helper.tsx:11]: enums as keys here

File: packages/ui/src/header/helper.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    "relative flex w-full flex-shrink-0 flex-row items-center justify-between gap-x-2 gap-y-4 bg-custom-sidebar-background-100",
+  secondary: "block !py-0  overflow-y-hidden border-b border-custom-border-200",
+  ternary: "flex justify-between py-2  border-b border-custom-border-200",
+};
+export const minHeights: IHeaderProperties = {
+  primary: "",
```
Comment (rahulramesha) [inline packages/ui/src/header/helper.tsx:17]: same here

File: packages/ui/src/row/helper.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+export type TRowVariant = "regular" | "hugging";
```
Comment (rahulramesha) [inline packages/ui/src/row/helper.tsx:1]: same as Variant enum, use the enum defined below here instead

File: packages/ui/src/row/helper.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+export enum ERowVariant {
+  regular = "regular",
+  hugging = "hugging",
+}
+export interface IRowProperties {
+  [key: string]: string;
+}
+export const rowStyle: IRowProperties = {
+  regular: "px-page-x",
```
Comment (rahulramesha) [inline packages/ui/src/row/helper.tsx:10]: here as well

File: web/app/[workspaceSlug]/(projects)/profile/[userId]/navbar.tsx [hunk 7]
```diff
@@ -20,7 +21,7 @@ export const ProfileNavbar: React.FC<Props> = (props) => {
   const tabsList = isAuthorized ? [...PROFILE_VIEWER_TAB, ...PROFILE_ADMINS_TAB] : PROFILE_VIEWER_TAB;
 
   return (
-    <div className="sticky -top-0.5 hidden md:flex items-center justify-between gap-4 border-b border-custom-border-300 bg-custom-background-100 px-4 sm:px-5 md:static">
+    <CustomHeader variant="secondary" className="sticky -top-0.5 hidden md:flex md:static">
```
Comment (rahulramesha) [inline web/app/[workspaceSlug]/(projects)/profile/[userId]/navbar.tsx:24]: please use the enum that was created for variant

File: web/core/components/issues/issue-layouts/filters/applied-filters/roots/cycle-root.tsx [hunk 8]
```diff
@@ -76,15 +77,16 @@ export const CycleAppliedFiltersRoot: React.FC = observer(() => {
...
-    <div className="flex justify-between p-4 gap-2.5">
-      <AppliedFiltersList
-        appliedFilters={appliedFilters}
-        handleClearAllFilters={handleClearAllFilters}
-        handleRemoveFilter={handleRemoveFilter}
-        labels={projectLabels ?? []}
-        states={projectStates}
-      />
-
+    <CustomHeader variant="ternary">
```
Comment (rahulramesha) [inline web/core/components/issues/issue-layouts/filters/applied-filters/roots/cycle-root.tsx:80]: here as well, please check and replace wherever equiured

**Review by rahulramesha — CHANGES_REQUESTED** [review]

File: packages/ui/src/row/row.tsx (hunk 1 above)
Comment (gakshita) [reply packages/ui/src/row/row.tsx:11]: This format was followed in the button component, thats why i have kept it this way.

**Review by rahulramesha — APPROVED** [review]

**Pull request merged**